src/main/java/com/spikes2212/command/drivetrains/smartmotorcontrollerdrivetrain/SparkTankDrivetrain.java

```diff
@@ -147,17 +146,25 @@ public void configureLoop(PIDSettings leftPIDSettings, PIDSettings rightPIDSetti
      * @param rightSetpoint            the right side loop's target setpoint
      * @param leftPIDSettings          the left side's PID constants
      * @param rightPIDSettings         the right side's PID constants
-     * @param feedForwardSettings      the feed forward gains
+     * @param leftFeedForwardSettings      the feed forward gains
      * @param trapezoidProfileSettings the trapezoid profile settings
      */
     @Override
     public void pidSet(UnifiedControlMode controlMode, double leftSetpoint, double rightSetpoint,
                        PIDSettings leftPIDSettings, PIDSettings rightPIDSettings,
-                       FeedForwardSettings feedForwardSettings, TrapezoidProfileSettings trapezoidProfileSettings) {
-        configPIDF(leftPIDSettings, rightPIDSettings, feedForwardSettings);
-        configureTrapezoid(trapezoidProfileSettings);
-        leftMaster.getPIDController().setReference(leftSetpoint, controlMode.getSparkMaxControlType());
-        rightMaster.getPIDController().setReference(rightSetpoint, controlMode.getSparkMaxControlType());
+                       FeedForwardSettings leftFeedForwardSettings, FeedForwardSettings rightFeedForwardSettings,
+                       TrapezoidProfileSettings leftTrapezoidProfileSettings,
+                       TrapezoidProfileSettings rightTrapezoidProfileSettings) {
+        configPIDF(leftPIDSettings, rightPIDSettings, leftFeedForwardSettings, rightFeedForwardSettings);
+        configureTrapezoid(leftTrapezoidProfileSettings);
+        FeedForwardController leftFeedForwardController = new FeedForwardController(leftFeedForwardSettings,
+                FeedForwardController.DEFAULT_PERIOD);
+        FeedForwardController rightFeedForwardController = new FeedForwardController(rightFeedForwardSettings,
+                FeedForwardController.DEFAULT_PERIOD);
```

Comment:
Make these fields instead of local variables. Remember that constructors should never receive a controller as direct input.